### ISSUE TITLE
JSON storage is too slow.  The old pickle was fine.

### ISSUE THREAD

**Issue by RalphCorderoy** [rss2email/rss2email#6]
The record of what had been seen in RSS feeds used to be in Python's pickle format and I never noticed a speed problem. When I switched to a distro using wking the slowdown was dramatic and the move to trendy JSON seems to be the problem.

Using `strace -ttt -fe open r2e list >/dev/null` I see 1.659636 s between the first open to start the CPython executable and opening ~/.local/share/rss2email.json, and then another 1.889195 s before it has finished the simple `list`. My JSON file is 19,543 lines, 919067 bytes.

I suggest targeting a reduction in `perf stat -Be instructions:u r2e list >/dev/null` by trying pickle format again. It's semi-textual, and available with every Python.

**Comment by jsbackus** [comment]
Hi @RalphCorderoy, sorry for taking so long to get back to you. I agree that JSON is too slow for large DBs, however, there are plenty of people who like it for its ease of maintenance. I believe a better option is to support multiple formats (controlled via config file) with the ability to switch back and forth. While pickle is faster than JSON, it still involves loading the entire database and so will become a problem eventually. So for really large databases, something like SQLite would be a better option.

So, my proposal is that we support the following formats:
* JSON
* Pickle
* SQLite / generic SQL

and that we add a new subcommand that allows the user to convert between the formats.

I'll update the name of the issue to reflect this.

**Comment by Ekleog** [comment]
I don't think we should support multiple database formats, that's just way too complex. However, switching to SQLite would probably be for the best, thanks to the ACID guarantees, so let's rename this issue to that :)

**Comment by RalphCorderoy** [comment]
As the person who opened this issue, I would be happy with just SQLite support.
